exercises/bracket-push/example.rs

```diff
@@ -0,0 +1,75 @@
+use std::collections::HashMap;
+
+pub struct Brackets {
+    raw_brackets: Vec<char>,
+    pairs: MatchingBrackets,
+}
+
+impl<'a> From<&'a str> for Brackets {
+    fn from(i: &str) -> Self {
+        Brackets::new(String::from(i), None)
+    }
+}
+
+#[derive(Clone)]
+pub struct MatchingBrackets {
+    collection: HashMap<char, char>,
+}
+
+impl From<Vec<(char, char)>> for MatchingBrackets {
+    fn from(i: Vec<(char, char)>) -> Self {
+        let mut c: HashMap<char, char> = HashMap::new();
+
+        for (k, v) in i {
+            c.insert(k, v);
+        }
+
+        MatchingBrackets { collection: c }
+    }
+}
+
+impl MatchingBrackets {
+    fn contains(&self, other: &char) -> bool {
+        let known = self.collection.keys().chain(self.collection.values()).collect::<Vec<_>>();
+        known.contains(&other)
+    }
+
+    fn closer_for(&self, k: &char) -> Option<&char> {
+        self.collection.get(k)
+    }
+}
+
+impl Brackets {
+    pub fn new(s: String, pairs: Option<Vec<(char, char)>>) -> Self {
+        let p = match pairs {
+            Some(x) => MatchingBrackets::from(x),
+            None => MatchingBrackets::from(vec![('[', ']'), ('{', '}'), ('(', ')')]),
+        };
+
+        Brackets {
+            raw_brackets: s.chars().filter(|c| p.contains(&c)).collect::<Vec<char>>(),
+            pairs: p.clone(),
```

Comment:
Huh. Now it's not angry. I must have been doing something wrong or fixed something without knowing.